src/challenges.js

```diff
@@ -94,9 +189,18 @@ const limitFunctionCallCount = (cb, n) => {
 // console.log(cube.checkIfCube());  // "We have a cube!"
 
 // Challenge 3: Recursion
-const checkMatchingLeaves = obj => {
-  // return true if every property on `obj` is the same
-  // otherwise return false
+const checkMatchingLeaves = (obj, comparator) => {
```

Comment:
Wow, your solution is really awesome! When I did this, I used recursion from a second helper function but you did yours with one!